v8isolate.cc

```diff
@@ -14,14 +14,31 @@ void* ArrayBufferAllocator::AllocateUninitialized(size_t length) {
 
 void ArrayBufferAllocator::Free(void* data, size_t) { free(data); }
 
-V8Isolate::V8Isolate() {
+V8Isolate::V8Isolate() : startup_data{} {
   v8::Isolate::CreateParams create_params;
   create_params.array_buffer_allocator = &allocator;
   isolate_ = v8::Isolate::New(create_params);
 }
 
+V8Isolate::V8Isolate(const char* snapshot_js_str) {
+  startup_data = v8::V8::CreateSnapshotDataBlob(snapshot_js_str);
+  if (startup_data.data == NULL) {
+    throw BadSnapshotDataException();
```

Comment:
So there's no way to detect whether the JS in your snapshot initialization succeeded? That's horrible.